x-pack/plugin/sql/sql-client/src/test/java/org/elasticsearch/xpack/sql/client/VersionTests.java

```diff
@@ -31,27 +31,74 @@ public void testInvalidVersion() {
         assertEquals("Invalid version format [7.1]", err.getMessage());
     }
 
-    public void testVersionFromJarInJar() throws IOException {
+
+    private static final String JAR_PATH_SEPARATOR = "!/";
+
+    private static String versionString(byte[] parts) {
+        StringBuffer version = new StringBuffer();
+        for (byte part : parts) {
+            if (version.length() > 0) {
+                version.append(".");
+            }
+            version.append(part);
+        }
+        return version.toString();
+    }
+
+    private static Path createDriverJar(byte[] parts) throws IOException {
         final String JDBC_JAR_NAME = "es-sql-jdbc.jar";
-        final String JAR_PATH_SEPARATOR = "!/";
 
         Path dir = createTempDir();
-        Path jarPath = dir.resolve("uberjar.jar");          // simulated uberjar containing the jdbc driver
-        Path innerJarPath = dir.resolve(JDBC_JAR_NAME); // simulated ES JDBC driver file
+        Path jarPath = dir.resolve(JDBC_JAR_NAME); // simulated ES JDBC driver file
 
         Manifest jdbcJarManifest = new Manifest();
         Attributes attributes = jdbcJarManifest.getMainAttributes();
         attributes.put(Attributes.Name.MANIFEST_VERSION, "1.0.0");
         attributes.put(new Attributes.Name("Change"), "abc");
-        attributes.put(new Attributes.Name("X-Compile-Elasticsearch-Version"), "1.2.3");
+        attributes.put(new Attributes.Name("X-Compile-Elasticsearch-Version"), versionString(parts));
 
         // create the jdbc driver file
-        try (JarOutputStream jdbc = new JarOutputStream(Files.newOutputStream(innerJarPath, StandardOpenOption.CREATE), jdbcJarManifest)) {}
+        try (JarOutputStream __ = new JarOutputStream(Files.newOutputStream(jarPath, StandardOpenOption.CREATE), jdbcJarManifest)) {}
+
+        return jarPath;
+    }
+
+    public void testVersionFromFileJar() throws IOException {
+        byte[] parts = {0, 1, 2};
+        Path jarPath = createDriverJar(parts);
+
+        URL fileUrl = new URL(jarPath.toUri().toURL().toString());
+        SqlVersion version = ClientVersion.extractVersion(fileUrl);
+
+        assertEquals(parts[0], version.major);
+        assertEquals(parts[1], version.minor);
+        assertEquals(parts[2], version.revision);
+        assertEquals(versionString(parts), version.version);
+    }
+
+    public void testVersionFromJar() throws IOException {
+        byte[] parts = {1, 2, 3};
```

Comment:
Why is this `{1, 2, 3}` and the previous test is `{0, 1, 2}`. It is relevant for these tests to have different versions to test? If not, and for the sake of using whatever versions, can you use a random value?

Reply:
Right, randomising it is a better idea. I've added that, thanks.